src/Player.js

```diff
@@ -0,0 +1,79 @@
+function Player(name) {
+  this.name = name
+  this.bowls = []
+  this.frameCount =1
+  this.score = 0
+  this.tenthIsStrike = false
+  this.tenthIsSpare = false
+  this.gameOver = false
+}
+
+Player.prototype.addFrame = function(one, two) {
+if (this.frameCount < 10) {
+  if(one <= 9) {
+    this.bowls.push(one);
+    this.bowls.push(two);
+  } else if( one == 10) {
+    this.bowls.push(one);
+  } else {
+    this.bowls.push(0);
+    this.bowls.push(0);
+  }
+} else if (this.frameCount === 10) {
+  this.SetTenthIsStrike(one);
+  if (one < 10) {
+    this.SetTenthIsSpare(one,two);
+  }
+  if (this.tenthIsStrike) {
+    this.bowls.push(one);
+  } else if (this.tenthIsSpare) {
+    this.bowls.push(one);
+    this.bowls.push(two);
+  } else {
+    this.bowls.push(one);
+    this.bowls.push(two);
+    this.gameOver = true;
+    this.calcScore();
+  }
+} else if (this.frameCount <= 11) {
+  if(this.tenthIsStrike) {
+    this.bowls.push(one);
+    this.bowls.push(two);
+    this.gameOver = true;
+    this.calcScore();
+  } else if(this.tenthIsSpare) {
+    this.bowls.push(one);
+    this.gameOver = true;
+    this.calcScore();
+  }
+}
+  this.frameCount += 1;
+};
```

Comment:
^could this be shorter>

Reply:
Yeah, I probably need to extract the two ways that the scores are pushed, either just bowl one pushed or both bowl one and bowl two, then maybe I could remove the tenth is strike method and just have an is strike method and use that with an if statement to set the value of the tenthIsStrike/Spare boolean attribute

Reply:
Yes. But even as a small refactor you could have a local variable to store whether the tenth frame is a strike, since you only use it in one function.